frontend/src/employee-frontend/components/holiday-term-periods/TermPeriodEditor.tsx

```diff
@@ -26,13 +33,25 @@ export default React.memo(function TermPeriodEditor() {
   return (
     <Container>
       <ContentArea opaque>
-        {renderResult(preschoolTerms, (preschoolTerms) => (
-          <PreschoolTermForm
-            allTerms={preschoolTerms}
-            onSuccess={navigateToList}
-            onCancel={navigateToList}
-          />
-        ))}
+        {termId === 'new'
+          ? renderResult(preschoolTerms, (preschoolTerms) => (
+              <PreschoolTermForm
+                allTerms={preschoolTerms}
+                onSuccess={navigateToList}
+                onCancel={navigateToList}
+              />
+            ))
+          : renderResult(
+              combine(preschoolTerms, preschoolTerm),
```

Comment:
Seems unnecessary to fetch all and also fetch one separately again. We can always find the one that is being edited from the list which we anyway need to fetch, so the whole fetch by id endpoint is unnecessary even on backend.